AdvancedVehicleOptions/GUI/UIOptionPanel.cs

```diff
@@ -566,8 +566,8 @@ private void SetupControls()
 
                 VehicleOptions.prefabUpdateEngine = m_options.prefab;
                 VehicleOptions.prefabUpdateUnits = m_options.prefab;
-                new EnumerableActionThread(VehicleOptions.UpdateBackEngines);
-                new EnumerableActionThread(VehicleOptions.UpdateCapacityUnits);
+                SimulationManager.instance.AddAction(VehicleOptions.UpdateBackEngines);
+                SimulationManager.instance.AddAction(VehicleOptions.UpdateCapacityUnits);

```

Comment:
The same story is here. If we don't care about result immediately then no issue.